test_result.go

```diff
@@ -0,0 +1,405 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/iver-wharf/wharf-api/internal/ctxparser"
+	"github.com/iver-wharf/wharf-core/pkg/ginutil"
+	"github.com/iver-wharf/wharf-core/pkg/problem"
+	"gorm.io/gorm"
+)
+
+type buildTestResultModule struct {
+	Database *gorm.DB
+}
+
+type artifactListItem struct {
+	FileName   string `json:"fileName"`
+	ArtifactID uint   `json:"artifactId"`
+}
+
+// TestResultListSummary contains data about several test result files.
+type TestResultListSummary struct {
+	BuildID uint `json:"buildId"`
+	Total   uint `json:"total"`
+	Failed  uint `json:"failed"`
+	Passed  uint `json:"passed"`
+	Skipped uint `json:"skipped"`
+}
```

Comment:
Most of your types here are exported, but not `artifactListItem`. Suggest exporting it as well (capitalizing it) for consistency.